onnxruntime/core/graph/contrib_ops/reverse_sequence_schema_defs.cc

```diff
@@ -57,6 +57,40 @@ static const char* Input_Data_Format_ver1_doc = R"DOC(
 or batch major (e.g. [batch_size, seq_length, ...]). Must be one of time_major (default), or batch_major.
 )DOC";
 
+void ReverseSequenceShapeInference(InferenceContext& ctx) {
+  propagateElemTypeFromInputToOutput(ctx, 0, 0);
+  if (!hasNInputShapes(ctx, 2)) {
+    return;
+  }
+
+  auto data_format = getAttribute(ctx, "data_format", "time_major");
+  auto& first_input_shape = getInputShape(ctx, 0);
+  if (first_input_shape.dim_size() < 2) {
+    fail_shape_inference("First input tensor must have rank >= 2");
+  }
+
+  TensorShapeProto::Dimension batch_size;
+  if (data_format == "time_major") {
+    batch_size = first_input_shape.dim(1);
+  } else {
+    batch_size = first_input_shape.dim(0);
+  }
+
+  if (batch_size.has_dim_value()) {
+    auto& seq_len_input_shape = getInputShape(ctx, 1);
+    auto batch_dim = seq_len_input_shape.dim(0);
+    if (batch_dim.has_dim_value()) {
+      if (static_cast<int64_t>(batch_dim.dim_value()) != static_cast<int64_t>(batch_size.dim_value())) {
+        fail_shape_inference("Batch size mismatch for input and sequence_lens.")
+      }
+    }
+
+    fail_shape_inference("Batch size must match for the first two inputs")
```

Comment:
line 88 should be removed. sorry. will update in next update

Reply:
Can we handle having a symbolic value for the batch size? I believe in that case (iirc) has_dim_param() would return true, and we could check the param name for a match.

Reply:
sorry I am not familiar with dim_param() comparing, so should batch_dim.dim_param() == batch_size.dim_param() always be True in this case? If we assume both are has_dim_param not dim_value.

Reply:
I would leave it as is. If at least one of them is symbolic, we cannot statically report an error. Two different symbols "M" and "N" does not mean that at runtime the values of these will differ … it is an error only if they differ, and that can only be caught at runtime.

Reply:
@gramalingam  this is exactly my current understanding. thanks for the clarification.